libdevcore/RLP.h

```diff
@@ -63,415 +86,780 @@ template <class T> struct Converter { static T convert(RLP const&, int) { BOOST_
-	RLPs toList(int _flags = Strict) const;
-
-	/// @returns the data payload. Valid for all types.
-	bytesConstRef payload() const { auto l = length(); if (l > m_data.size()) BOOST_THROW_EXCEPTION(BadRLP()); return m_data.cropped(payloadOffset(), l); }
-
-	/// @returns the theoretical size of this item as encoded in the data.
-	/// @note Under normal circumstances, is equivalent to m_data.size() - use that unless you know it won't work.
-	size_t actualSize() const;
+    /// Conversion flags
+    enum
+    {
+        AllowNonCanon = 1,
+        ThrowOnFail = 4,
+        FailIfTooBig = 8,
+        FailIfTooSmall = 16,
+        Strict = ThrowOnFail | FailIfTooBig,
+        VeryStrict = ThrowOnFail | FailIfTooBig | FailIfTooSmall,
+        LaissezFaire = AllowNonCanon
+    };
+
+    using Strictness = int;
+
+    /// Construct a null node.
+    RLP() {}
+
+    /// Construct a node of value given in the bytes.
+    explicit RLP(bytesConstRef _d, Strictness _s = VeryStrict);
+
+    /// Construct a node of value given in the bytes.
+    explicit RLP(bytes const& _d, Strictness _s = VeryStrict) : RLP(&_d, _s) {}
+
+    /// Construct a node to read RLP data in the bytes given.
+    RLP(byte const* _b, unsigned _s, Strictness _st = VeryStrict)
+      : RLP(bytesConstRef(_b, _s), _st)
+    {}
+
+    /// Construct a node to read RLP data in the string.
+    explicit RLP(std::string const& _s, Strictness _st = VeryStrict)
+      : RLP(bytesConstRef((byte const*)_s.data(), _s.size()), _st)
+    {}
+
+    /// The bare data of the RLP.
+    bytesConstRef data() const { return m_data; }
+
+    /// @returns true if the RLP is non-null.
+    explicit operator bool() const { return !isNull(); }
+
+    /// No value.
+    bool isNull() const { return m_data.size() == 0; }
+
+    /// Contains a zero-length string or zero-length list.
+    bool isEmpty() const
+    {
+        return !isNull() && (m_data[0] == c_rlpDataImmLenStart ||
+                                m_data[0] == c_rlpListStart);
+    }
+
+    /// String value.
+    bool isData() const { return !isNull() && m_data[0] < c_rlpListStart; }
+
+    /// List value.
+    bool isList() const { return !isNull() && m_data[0] >= c_rlpListStart; }
+
+    /// Integer value. Must not have a leading zero.
+    bool isInt() const;
+
+    /// @returns the number of items in the list, or zero if it isn't a list.
+    size_t itemCount() const { return isList() ? items() : 0; }
+    size_t itemCountStrict() const
+    {
+        if (!isList())
+            BOOST_THROW_EXCEPTION(BadCast());
+        return items();
+    }
+
+    /// @returns the number of bytes in the data, or zero if it isn't data.
+    size_t size() const { return isData() ? length() : 0; }
+    size_t sizeStrict() const
+    {
+        if (!isData())
+            BOOST_THROW_EXCEPTION(BadCast());
+        return length();
+    }
+
+    /// Equality operators; does best-effort conversion and checks for equality.
+    bool operator==(char const* _s) const
```

Comment:
I though there was an option to allow one-liners for inline function implementations? Or was this wider that 80 symbols?

Reply:
It does not fit (82 chars).